pytorch_lightning/loggers/wandb.py

```diff
@@ -104,9 +104,7 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
 
     @rank_zero_only
     def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
-        if step is not None:
-            metrics['global_step'] = step
```

Comment:
I think the intention here was to set the step to the global step if None was passed in, so it should be
`step = metrics['global_step']
`and this should solve the bug

Reply:
That was also my idea. Maybe in a previous version of wandb there was no step parameter in the log method. But simply add this to metrics dict will effect other loggers and is currently not necessary.

Reply:
no, what I'm saying is that the reason why it affects other loggers is because there you have currently.
`metrics['global_step'] = step `
But it should be 
`step = metrics['global_step']`
and this is the actual fix for the bug, not deleting it. If no specific step is provided, it should just take the global step, right? Or am I missing something?

Reply:
I guess it depends if we want to set the step explicitly when None is provided, or let wandb track the global step internally. I guess either is fine. Just a detail to consider.
So either `step = metrics['global_step']` or delete it like you did now. But yeah, definitely not `metrics['global_step'] = step`

Reply:
We should check how the other loggers handle this and make it consistent

Reply:
looks like most other loggers don't check for the None and resolve that internally to the last iteration/global step. So I guess your fix is fine and so you should update the test.